ui/ui-components/pages/model/[modelId]/source/[sourceId]/edit.tsx

```diff
@@ -201,8 +201,8 @@ const Page: NextPage<Props> = ({
   }, [preview]);
 
   const mappingColumn = useMemo(
-    () => Object.keys(source.mapping)[0] ?? previewColumns?.[0],
-    [previewColumns, source.mapping]
+    () => Object.keys(source.mapping)[0],
+    [source.mapping]
```

Comment:
I _think_ this gets used downline to set a default value with `columnName`, so we might be able to get rid of it. `mappingPropertyKey` as well perhaps.